torch/csrc/jit/mobile/debug_info.cpp

```diff
@@ -34,19 +102,45 @@ MobileDebugTable::MobileDebugTable(
         if (tup_elems.size() == 3) {
           int64_t debug_handle = tup_elems[1].toInt();
           auto source_range = deserializer->deserialize(tup_elems[2]);
-          source_range_map_.emplace(debug_handle, std::move(source_range));
+          source_range_map.emplace(debug_handle, std::move(source_range));
         }
       }
     }
   }
+  const std::string callstack_debug_file("callstack_debug_map.pkl");
+  if (reader->hasRecord("callstack_debug_map.pkl")) {
+    at::DataPtr callstack_data;
+    size_t callstack_data_size{0};
+    std::tie(callstack_data, callstack_data_size) =
+        reader->getRecord(callstack_debug_file);
+    InlinedCallStackUnpickler unpickler;
+    callstack_ptr_map_ = unpickler.unpickle(
```

Comment:
@raziel The reason it is hard to decouple InlinedCallStackUnpickler from source range info is that the unpickling returns a map of InlinedCallStackPtrs and InlinedCallStack contains pointers to SourceRange and ClassType information.

So I understand the need to keep serialization/deserialization separate from interpreter but this dependency is only one way. That is MobileDebugTable depends on interpreter to provide compilation unit.

But I do see your point about dependency of MobileDebugTable on interpreter. In order for us to deserialize debug info now there is a dependency on interpreter's import.cpp, but I feel that might be ok?

Or is there some other dependency you are worried about?

Reply:
> So, the question is whether InlinedCallStack data should be part of the core functionality or it belongs only if we're going to run things through the interpreter.

I would say that InlinedCallStack should be part of the debug info regardless of whether we run through interpreter or not.

Now regarding the question of dependency on lite interpreter for deserialization of callstack_debug_map: So while we expect compilation_unit in order to lookup class_types, which are part of ModuleInstanceInfo, it is strictly not necessary for symbolication since we will stringify class_type anyway. And serialized class_types are strings. We definitely need to handle the case where compilation_unit does not have class_type. This is because when we lower some submodule or entire module to backend, all class types which are subsumed in that module are lost. For example if entire module is lowered then no class types will be found in compilation unit. In the last PR in the stack I handled this by just overloading instance name with class type. Given the discussion here it makes sense to resort to that as the de facto method and remove dependency on compilation_unit entirely.

> For example, if a model runs entirely delegated to a backend, do we still have InlinedCallStack data being serialized? Which should? be the case since there will still be some instructions to orchestrate the delegation, which I assume will have a call stack data.

Right. So if module is entirely run in backend then InlinedCallStack will have stack info corresponding to interpreter instructions that orchestrate execution and rest of the stack will correspond to what was lowered. This is much clearer in the last PR.